lib/mayaUsd/listeners/proxyShapeNotice.h

```diff
@@ -27,11 +27,11 @@ PXR_NAMESPACE_OPEN_SCOPE
 class MayaUsdProxyShapeBase;
 
 /// Notice sent when the ProxyShape loads a new stage
-class UsdMayaProxyStageSetNotice : public TfNotice
+class MayaUsdProxyStageBaseNotice : public TfNotice
```

Comment:
Renamed this one "UsdMaya" -> "MayaUsd" and made it a base notice class. Below created this existing one (deriving from this base) and new one for stage invalidate.